src/Http/Client.php

```diff
@@ -227,6 +227,32 @@ public function addCookie(CookieInterface $cookie)
         return $this;
     }
 
+    /**
+     * Client instance returned is scoped to the domain, port, and scheme parsed from the passed URL string. The passed
+     * string must have a scheme and a domain. Optionally, if a port is included in the string, the port will be scoped
+     * too. Other parts of the url string are ignored.
+     *
+     * @param  string $url A string URL e.g. https://example.com
+     * @return static
+     * @throws \InvalidArgumentException
+     */
+    public static function createScopedClientFromUrl(string $url)
```

Comment:
I don't think using "scoped" in the name is necessary and the "client" is also redundant. The name could be shortened to `createFromUrl()`.